Adding a test for char-s2s

tomekd opened the pull request.

tomekd committed:
Test char-s2s model

snukky committed:
Fix & update char-s2s test

snukky committed:
Fix MRT_MARIAN_USE_CUDNN

snukky commented:
Have you run your test? I think it couldn't have work. I fixed `translation.yml` to some extend; now it's here: `path/to/data.statmt.org/romang/marian-regression-tests`. 

The error I got:
```
Requested shape for existing parameter 'F0::encoder_0_highway_d1_W' does not match original shape
Aborted from marian::Expr marian::ExpressionGraph::param(std::__cxx11::string, marian::Shape, Args ...) [with Args = {marian::keywords::Keyword<3329549428u, std::function<void(std::shared_ptr<marian::TensorBase>)> >}; marian::Expr = std::shared_ptr<marian::Chainable<std::shared_ptr<marian::TensorBase> > >; std::__cxx11::string = std::__cxx11::basic_string<char>] in /home/romang/work/marian/marian-dev/src/graph/expression_graph.h: 275
```
Please, fix it keeping the minimum possible number of options in it.

Also add g+w to the marian folder you have created.

BTW, I also found:
```
.../marian-dev/src/tests/conv_test.cu(52): error: no instance of constructor "marian::Convolution::Convolution" matches the argument list
            argument types are: (const char [12], int, int, int)
```

tomekd commented:
Hi,
I did run the tests. This is really strange. Taking a look at that now.

tomekd commented:
1. I found the problem. I did not change the translate.yml file on vali but locally. 
2. I added g+w on vali as you wished.
3. The compilation error is with a rubbish test that should be there. I fixed and jenkins confirmed that.
4. I can't reproduce the problem with ``F0::encoder_0_highway_d1_W``. Have you tried to clean marian build and rebuild from the scratch?

tomekd committed:
change the name of the model
